### PR TITLE
Fix IsSafeFileToDownload (thx [WPMG]PRoSToTeM@)

### PR DESCRIPTION
**Pull request by In-line**
* Fix nullptr dereferencing
* Remove redundant check 
* Fix comment

### CODE REVIEW

File: rehlds/engine/sv_main.cpp [hunk 1]
```diff
@@ -7327,17 +7327,22 @@ qboolean IsSafeFileToDownload(const char *filename)
...
+	else
+		last = nullptr;
 #else
 	last = Q_strrchr(lwrfilename, '.');
 #endif
 
 	if (lwrfilename[0] == '/'
 		|| Q_strstr(lwrfilename, "\\")
 		|| Q_strstr(lwrfilename, ":")
+#ifndef REHLDS_FIXES // Redundant check
```
Comment (theAsmodai) [inline rehlds/engine/sv_main.cpp]: Why?

**Comment by LevShisterov** [comment]
This function should be rewritten from scratch, IMHO. I see no reason to limit ext to 3 chars, require ext, etc..

File: rehlds/engine/sv_main.cpp (hunk 1 above)
Comment (In-line) [inline rehlds/engine/sv_main.cpp]: Later we check dot count to be 1 (I added comment about it)

**Comment by In-line** [comment]
@LevShisterov Are you sure it is safe to do that ?

**Comment by LevShisterov** [comment]
Depends on how good it will be done.

File: rehlds/engine/sv_main.cpp (hunk 1 above)
Comment (theAsmodai) [inline rehlds/engine/sv_main.cpp]: Ok, but such things must be commented

**Comment by IgnacioFDM** [comment]
We would definitely need some very extensive unit-testing coded for this function if any big changes were to happen

File: rehlds/engine/sv_main.cpp (hunk 1 above)
Comment (In-line) [inline rehlds/engine/sv_main.cpp]: Okey. I will later do that

**Comment by In-line** [comment]
@LevShisterov We can add `// TODO: ...` to the code about it.

**Pull request merged**